Fix breaking on HTML email without images

dagobert opened the pull request:
## Breaking Change:

no

## Description:

If using html emails with no images the code breaks since it is not tested for empty (uninitialized) key images.

**Related issue (if applicable):** fixes #22142

## Example entry for `configuration.yaml` (if applicable):
```yaml
- alias: notify_test
  initial_state: 'on'
  trigger:
      platform: time_pattern
      minutes: '/2'
  action:
      service: notify.smtp
      data_template:
          title: >
              [ShuFu] Test
          message: |
              Test text
          data:
              html: >
                  <!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//DE" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
                  <html lang="de" xmlns="http://www.w3.org/1999/xhtml">
                  <head>
                  </head>
                  <body>
                  TEST html
                  </body>
                  </html>
```

## Checklist:
  - [x ] The code change is tested and works locally.
  - [ ] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x ] There is no commented out code in this PR.

dagobert committed:
Fix breaking on HTML email without images

If using html emails with no images the code breaks since it is not tested for empty (uninitialized) key images.

dagobert committed:
fixed long line

cgtobi commented on homeassistant/components/notify/smtp.py:
```diff
@@ -241,14 +241,16 @@ def _build_html_msg(text, html, images):
-        except FileNotFoundError:
-            _LOGGER.warning("Attachment %s [#%s] not found. Skipping",
-                            atch_name, atch_num)
+    if images is not None:
```
This should be enough.
```python
if images:
```

dagobert replied:
It is best practice and recommended to use `is not None` (see [Python docs](https://www.python.org/dev/peps/pep-0008/#programming-recommendations))

> Comparisons to singletons like None should always be done with `is` or `is not`, never the equality operators.
>
> Also, beware of writing `if x` when you really mean `if x is not None` -- e.g. when testing whether a variable or argument that defaults to None was set to some other value. The other value might have a type (such as a container) that could be `false` in a boolean context!

cgtobi replied:
Thanks for pointing that out.

awarecan commented:
You can change line 157 to

```python
message, images=data.get(ATTR_IMAGES, []))
```

dagobert commented:
> You can change line 157 to
> 
> ```python
> message, images=data.get(ATTR_IMAGES, []))
> ```

Thank for this tip. Unfortunately you referenced the wrong line. I had to change line 154 . 157 has / should stay the same since it covers the option of having images without html. If we change it also in that line we allow data key without any subkey such as html or images. That could be the of the main preference developers to avoid errors in the log but imho it is not necessary because it does not make to have data without any subkey.

I will change the pull request.

dagobert committed:
Implemented suggested better solution

Better solution to allow data -> html email without images.

awarecan commented:
I think we better guard both HTML and multi part message usage.

dagobert committed:
Protecting data -> without html key from crashing

If the data key does not contain the html key, sending the email would crash this "script". Preventing this by returning an default empty array.

dagobert commented:
> I think we better guard both HTML and multi part message usage.

After rethinking it, I agree.
